cpp/path_module/algorithm/path.cpp

```diff
@@ -1,301 +1,343 @@
         break;
       case '+':
         label_string.remove_prefix(1);
-        labelSets.whitelist.insert(label_string);
+        config_.label_sets.whitelist.insert(label_string);
         break;
       case '/':
         label_string.remove_prefix(1);
-        labelSets.termination_list.insert(label_string);
+        config_.label_sets.termination_list.insert(label_string);
         break;
       default:
-        labelSets.whitelist.insert(label_string);
+        config_.label_sets.whitelist.insert(label_string);
         break;
     }
   }
 }
 
 /*function that takes input list of relationships, and sorts them into appropriate categories
 sets were also used to reduce complexity*/
-void Path::ParseRelationships(const mgp::List &list_of_relationships, RelationshipSets &relationshipSets,
-                              bool &any_outgoing, bool &any_incoming) {
+void Path::PathHelper::ParseRelationships(const mgp::List &list_of_relationships) {
   if (list_of_relationships.Size() ==
       0) {  // if no relationships were passed as arguments, all relationships are allowed
-    any_outgoing = true;
-    any_incoming = true;
+    config_.any_outgoing = true;
+    config_.any_incoming = true;
     return;
   }
+
   for (const auto rel : list_of_relationships) {
-    std::string rel_type = std::string(rel.ValueString());
-    const size_t size = rel_type.size();
-    const char first_elem = rel_type[0];
-    const char last_elem = rel_type[size - 1];
-    if (first_elem == '<' && last_elem == '>') {
-      throw mgp::ValueException("Wrong relationship format => <relationship> is not allowed!");
-    } else if (first_elem == '<' && size == 1) {
-      any_incoming = true;  // all incoming relatiomships are allowed
-    } else if (first_elem == '<' && size != 1) {
-      relationshipSets.incoming_rel.insert(rel_type.erase(0, 1));  // only specified incoming relationships are allowed
-    } else if (last_elem == '>' && size == 1) {
-      any_outgoing = true;  // all outgoing relationships are allowed
-
-    } else if (last_elem == '>' && size != 1) {
-      rel_type.pop_back();
-      relationshipSets.outgoing_rel.insert(rel_type);  // only specifed outgoing relationships are allowed
-    } else {                                           // if not specified, a relationship goes both ways
-      relationshipSets.any_rel.insert(rel_type);
+    std::string rel_type{std::string(rel.ValueString())};
+    bool starts_with = rel_type.starts_with('<');
+    bool ends_with = rel_type.ends_with('>');
+
+    if (rel_type.size() == 1) {
+      if (starts_with) {
+        config_.any_incoming = true;
+      } else if (ends_with) {
+        config_.any_outgoing = true;
+      } else {
+        config_.relationship_sets[rel_type] = RelDirection::kAny;
+      }
+      continue;
+    }
+
+    if (starts_with && ends_with) {
+      config_.relationship_sets[rel_type.substr(1, rel_type.size() - 2)] = RelDirection::kBoth;
```

Comment:
Why is here -2?

Reply:
Maybe add comment how this should look

Reply:
Added comments
<rel_type> -> size is original size - 2, remove the starting < and > at the end